src/hazmat/sieve.rs

```diff
@@ -140,14 +140,14 @@ impl<T: Integer> Sieve<T> {
             return false;
         }
 
-        // Set the new base.
-        // Should not overflow since `incr` is never greater than `incr_limit`,
-        // and the latter is chosen such that it doesn't overflow when added to `base`
-        // (see the rest of this method).
-        self.base = self
-            .base
-            .checked_add(&self.incr.into())
-            .expect("addition should not overflow by construction");
+        // Set the new base. This increment will not overflow unless the `Sieve` is misused and manipulated.
+        match self.base.checked_add(&self.incr.into()).into() {
```

Comment:
I wonder if it's worth doing this check in the constructor, because it's the only place where that could happen.

Reply:
`self.incr` changes over the course of the sieving and is initialized to `0` in the constructor so I don't think that can work.

Reply:
Oh yes, you're right. `incr` is ensured to be `<= incr_limit` just above, and `base + incr_limit` is ensured not to overflow just below, so the `None` branch is never reached without private attribute manipulation, is it?

Reply:
Correct: as long as the logic surrounding the `incr` and `incr_limit` is not changed, this will not overflow (modulo manipulating private attributes).